### DISCUSSION TITLE
How to write to a file in Android

### DISCUSSION THREAD

**Discussion by charlesschaefer** [tauri-apps/tauri#10325] [Q&A] [answered]
Hello,
I'm building an app with Tauri 2 in which I'll need to allow users to create a backup of their data. I've seen the filesystem plugin, but it says it is only supported on windows, mac and linux.l

Is there a way to write to a file that the user selects (through the plugin-dialog's save() function) on Android?

**Comment by charlesschaefer** [comment, marked as answer]
Just an update. I haven't discovered yet how to allow users to select a path to write the file, because tauri's dialog plugin doesn't work on Android. But I was able to write to a file (i.e. /storage/emulated/0/Documents/file.txt) using Rust's std library (`std::fs::write`).

Maybe this helps someone in the future.

**Reply by owjs3901** [reply]
Why doesn't tauri std lib implement this?

This works!

The code is used my project

```rust
#[tauri::command]
async fn save_files(name: &str, data: Vec<Vec<u8>>)-> Result<(), String> {
    let now = std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs();
    // create directory if not exists
    tokio::fs::create_dir_all("/storage/emulated/0/DCIM/InstaFeedMaker").await.unwrap();
    for (i, data) in data.iter().enumerate() {
        tokio::fs::write(format!("/storage/emulated/0/DCIM/InstaFeedMaker/{}-{}-{}.png", name, i, now), data).await.unwrap();
    }
    Ok(())
}
```

**Reply by owjs3901** [reply]
However, there is a problem that you cannot add picture to the album.
